### ISSUE TITLE
Proposal: use Senza effectively for better cluster management

### ISSUE THREAD

**Issue by linki** [zalando-incubator/kubernetes-on-aws#57]
# TL;DR

embrace all the senza features for better cluster management

# Problem

i'm wondering why we don't use the blue/green versions stuff from senza correctly

# current setup

* different clusters are considered the same app from senza's point of view
* different senza versions represent entirely different clusters for us
* senza traffic doesn't make sense at all as it would balance traffic between two entirely different clusters
* senza delete without a version would affect all clusters of an aws account
* upgrading the stack happens outside of senza

# proposal (simplyfied to only consider the masters)

* create a senza app called "my-cluster" (senza init)
* deploy the first set of masters as "v1" (senza create)
* use senza traffic to point 100% of traffic to the v1 masters
* when deploying a new kubernetes version:
  * deploy the second set of masters as "v2" (senza create)
  * use senza traffic to slowly migrate traffic to the v2 masters
  * drop the v1 masters (senza delete)

additional worker pools would be additional senza stacks. instead of blue/green deployment we could in place updates but don't have to. the senza traffic would be useless for workers.

# advantages

* use all the goodness of senza
* be safer
* use blue/green deployments for master components

# challenges

* etcd prefix needs to be /registry-stack-name instead of /registry-stack-version
* zack: create a new role for each new cluster
  * proposal:
    * a PowerUser role for a specific aws account allows access to all clusters in that aws account
    * OR make a specific Kubernetes-Admin role that allows access to clusters but not to the entire aws account
    * using zack to protected different clusters inside the same aws account is more difficult (we don't want a separate role for each cluster)

**Comment by hjacobs** [comment]
I think splitting up into separate stacks is the way to go (#45), this easily allows us to use the "immutable stack" approach and `senza traffic` for the master nodes.

**Comment by mikkeloscar** [comment]
Our setup has evolved a lot since this issue was created, it is no longer so relevant to use senza this way.
